src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel_UpdateUI.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.VisualStudio.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal partial class DocumentOutlineViewModel
+    {
+        private readonly record struct UIData(ExpansionOption ExpansionOption, SnapshotPoint? CaretPoint);
```

Comment:
when do we not have a caret point?

Reply:
The underlying helper we call says that is possible: [MapUpOrDownToBuffer](https://github.com/dotnet/roslyn/blob/54a50247a9f875916947815176dacadf2d7444ee/src/EditorFeatures/Core/Shared/Extensions/IBufferGraphExtensions.cs) though looking at the code I do not understand when this would ever return null. I can change to just assert a make this not-null.